db/schema.rb

```diff
@@ -55,6 +55,46 @@
     t.index ["creator_id"], name: "index_documents_on_creator_id"
   end
 
+  create_table "event_document_approveds", force: :cascade do |t|
+    t.bigint "document_id"
+    t.bigint "user_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["document_id"], name: "index_event_document_approveds_on_document_id"
+    t.index ["user_id"], name: "index_event_document_approveds_on_user_id"
+  end
+
+  create_table "event_document_publisheds", force: :cascade do |t|
+    t.bigint "document_id"
+    t.bigint "user_id"
+    t.string "review_state"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["document_id"], name: "index_event_document_publisheds_on_document_id"
+    t.index ["user_id"], name: "index_event_document_publisheds_on_user_id"
+  end
+
+  create_table "event_document_submitteds", force: :cascade do |t|
+    t.bigint "document_id"
+    t.bigint "user_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["document_id"], name: "index_event_document_submitteds_on_document_id"
+    t.index ["user_id"], name: "index_event_document_submitteds_on_user_id"
+  end
+
+  create_table "event_document_updateds", force: :cascade do |t|
+    t.bigint "document_id"
+    t.bigint "user_id"
+    t.json "before", default: {}
```

Comment:
Could we get this by looking at the previous event?

Reply:
Yeah, or save just the changeset. The consideration for saving both is that we can then clean up old updated-events, and every event can be rendered without doing a weird query.